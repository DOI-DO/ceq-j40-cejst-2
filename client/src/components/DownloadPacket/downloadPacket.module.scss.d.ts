declare namespace DownloadPacketModuleScssNamespace {
    export interface IDownloadPacketModuleScss {
        downloadBoxContainer: string;
        downloadBox: string;
        downloadBoxTextBox: string;
        downloadBoxTitle: string;
        dataSourceText: string;
        dataSourceButtonContainer: string;
        shapefileButtonContainer: string;
        downloadBoxButton: string;
        dataSourceButtonText: string;
        shapeFileText: string;
        shapeFileButtonText: string;
        newTagContainer: string;
        newtag: string;
        lastUpdated:string;
    }
  }

declare const DownloadPacketModuleScssModule: DownloadPacketModuleScssNamespace.IDownloadPacketModuleScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: DownloadPacketModuleScssNamespace.IDownloadPacketModuleScss;
  };

export = DownloadPacketModuleScssModule;
