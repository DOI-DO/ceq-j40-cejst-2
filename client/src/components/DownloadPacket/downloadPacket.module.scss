@use '../../styles/design-system.scss' as *;

@mixin baseButtonContainerStyles {
  align-self: center;

  a {
    text-decoration: none;
  }

  img {
    // This should be a global css-filter value as it's generated
    // from the primary color.
    filter: invert(15%) sepia(9%) saturate(5666%) hue-rotate(175deg) brightness(96%) contrast(95%);
  }
}

@mixin baseTextStyle {
  @include typeset('sans', 'xs', 3);
  @include u-margin-top(2);
  @include u-margin-bottom(2);
  
  span {
    font-style: italic;
  }
};

@mixin baseButtonTextStyle {
  padding-top: 4px;
  padding-left: 5px;
}

.downloadBoxContainer {
    color: whitesmoke;
    margin: auto;
    @include u-margin-top(2.5);

    .downloadBox {
      @include u-bg('blue-80v');
      border-radius: 6px 6px;
      // @include u-margin-top(3);

      .downloadBoxTextBox {
        padding: 25px 25px;
        display: flex;
        flex-direction: column;
        
        .downloadBoxTitle {
          @include typeset('sans', 'xs', 3);
          @include u-text('semibold');
        }
        
        // Data source text
        .dataSourceText {
          @include baseTextStyle();
        }
        // Data source button
        .dataSourceButtonContainer {
          @include baseButtonContainerStyles();

          .dataSourceButtonText {
            @include baseButtonTextStyle();
          }
        }
        
        // Shapefile text
        .shapeFileText {
          @include baseTextStyle();
          @include u-margin-top(7);
        }
        // Shapefile button
        .shapefileButtonContainer{
          @include baseButtonContainerStyles();

          .shapeFileButtonText {
            @include baseButtonTextStyle();
            @include u-margin-right(3);
          }
        }

      }
    }
  }

  .downloadBoxButton{
    @include u-bg('white');
    @include u-color('blue-80v');
    display: flex;
  }

  .newtag {
    @include u-text("blue-70v");
    @include u-bg("yellow-20v");
  }