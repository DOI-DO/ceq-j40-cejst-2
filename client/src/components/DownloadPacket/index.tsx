import DownloadPacket from './DownloadPacket';

export default DownloadPacket;
