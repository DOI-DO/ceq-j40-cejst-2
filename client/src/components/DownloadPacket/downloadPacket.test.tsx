import * as React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import DownloadPacket from './DownloadPacket';

describe('download packet component defined', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <DownloadPacket />
      </LocalizedComponent>,
  );

  it('checks if component renders', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});

