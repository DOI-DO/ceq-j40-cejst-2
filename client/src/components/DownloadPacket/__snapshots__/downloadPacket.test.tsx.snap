// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`download packet component defined checks if component renders 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="grid"
  >
    <div>
      <div>
        <div>
          <div>
            Files available for download
          </div>
          <div>
            Download the data sources used in the CEJST (.csv and .xlsx, 52MB unzipped). 
            <span>
              Last updated: 04/06/22 
            </span>
          </div>
          <div>
            <a
              data-cy="download-link"
              download=""
              href="//"
            >
              <button
                class="usa-button"
                data-testid="button"
                type="button"
              >
                <div>
                  <img
                    alt="download icon for download package"
                    src="test-file-stub"
                  />
                </div>
                <div>
                  Download data sources
                </div>
              </button>
            </a>
          </div>
          <div>
            <span
              class="usa-tag"
              data-testid="tag"
            >
              NEW
            </span>
            Download the shapefiles, along with a codebook (.cpg, .dbf, .prj, .shp and .shx, 740MB unzipped). 
            <span>
              Last updated: 04/06/22 
            </span>
          </div>
          <div>
            <a
              data-cy="shapefile-link"
              download=""
              href="//"
            >
              <button
                class="usa-button"
                data-testid="button"
                type="button"
              >
                <div>
                  <img
                    alt="download icon for download package"
                    src="test-file-stub"
                  />
                </div>
                <div>
                  Download shapefiles
                </div>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
